third_party/com_github_aws_cpp_sdk/aws_cpp_sdk.BUILD.bazel

```diff
@@ -0,0 +1,180 @@
+        "aws-cpp-sdk-core/source/http/curl/*.cpp",  # HTTP_CURL_CLIENT_SOURCE
+        "aws-cpp-sdk-core/source/utils/crypto/openssl/*.cpp",  # UTILS_CRYPTO_OPENSSL_SOURCE
+    ]) + select({
+        "@bazel_tools//src/conditions:windows": glob([
+            "aws-cpp-sdk-core/source/net/windows/*.cpp",  # NET_SOURCE
+            "aws-cpp-sdk-core/source/platform/windows/*.cpp",  # PLATFORM_WINDOWS_SOURCE
+        ]),
+        "//conditions:default": glob([
+            "aws-cpp-sdk-core/source/net/linux-shared/*.cpp",  # NET_SOURCE
+            "aws-cpp-sdk-core/source/platform/linux-shared/*.cpp",  # PLATFORM_LINUX_SHARED_SOURCE
+        ]),
+    }),
+    hdrs = [
+        "aws-cpp-sdk-core/include/aws/core/SDKConfig.h",
+    ] + glob([
+        "aws-cpp-sdk-core/include/aws/core/*.h",  # AWS_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/auth/*.h",  # AWS_AUTH_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/client/*.h",  # AWS_CLIENT_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/internal/*.h",  # AWS_INTERNAL_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/net/*.h",  # NET_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/http/*.h",  # HTTP_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/http/standard/*.h",  # HTTP_STANDARD_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/config/*.h",  # CONFIG_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/monitoring/*.h",  # MONITORING_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/platform/*.h",  # PLATFORM_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/*.h",  # UTILS_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/event/*.h",  # UTILS_EVENT_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/base64/*.h",  # UTILS_BASE64_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/crypto/*.h",  # UTILS_CRYPTO_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/json/*.h",  # UTILS_JSON_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/threading/*.h",  # UTILS_THREADING_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/xml/*.h",  # UTILS_XML_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/memory/*.h",  # UTILS_MEMORY_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/memory/stl/*.h",  # UTILS_STL_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/logging/*.h",  # UTILS_LOGGING_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/ratelimiter/*.h",  # UTILS_RATE_LIMITER_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/stream/*.h",  # UTILS_STREAM_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/external/cjson/*.h",  # CJSON_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/external/tinyxml2/*.h",  # TINYXML2_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/http/curl/*.h",  # HTTP_CURL_CLIENT_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/crypto/openssl/*.h",  # UTILS_CRYPTO_OPENSSL_HEADERS
+    ]),
+    defines = [
```

Comment:
Do these not go into a config.h file or similar?

Reply:
I'm not sure. This was copied straight out of https://github.com/tensorflow/io/blob/4d70341df05ddce09c257d4ecaaaba1873594105/third_party/aws-sdk-cpp.BUILD#L78-L94

I'd rather spend time on this after the feasibility of trimming down the AWS SDK to the Auth code has been demonstrated. Added a TODO in the first comment.

Reply:
I've moved this defines into a `write_file` rule to generate `aws-cpp-sdk-core/include/aws/core/SDKConfig.h`, with some platform specific defines.